frontend/server/src/Controllers/Tag.php

```diff
@@ -65,4 +65,35 @@ public static function getPublicTags() {
             'problemTag'
         );
     }
+
+    /**
+     * Return most frequent public tags of a certain level
+     *
+     * @return list<array{alias: string}>
+     */
+    public static function getFrequentTagsByLevel(
+        string $problemLevel
+    ): array {
+        return \OmegaUp\Cache::getFromCacheOrSet(
+            \OmegaUp\Cache::TAGS_LIST,
+            'frequent',
```

Comment:
```suggestion
            "level-{$problemLevel}",
```

todas las invocaciones a `\OmegaUp\Cache::getFromCacheOrSet` deben tener todos los parámetros aquí en el nombre de la llave.

de lo contrario lo que pasaría es:

* alguien pide el nivel de karel, esto se guarda con la llave "frequent"
* luego, alguien más pide el nivel de programación competitiva, y al momento de buscar en el cache, como se usó la llave "frequent", se va a regresar el nivel de karel!